lib/src/select/select.component.ts

```diff
@@ -13,43 +15,45 @@ import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
         }
     ]
 })
+
 export class SelectComponent implements ControlValueAccessor {
 
-    @Input() placeholder: string;
-    @Input() options: Array<string> = [];
-    @Input() disabled: boolean = false;
-    disabledAlpha = 0.4;
-    enabledAlpha = 1.0;
-    _value: string;
-    onChange: any = () => {
-    };
-    onTouched: any = () => {
-    };
-
-    get alpha() {
-        return this.disabled ? this.disabledAlpha : this.enabledAlpha;
-    }
+    @Input() placeholder: string = '';
 
-    get value() {
-        return this._value;
-    }
+    _disabled: boolean = false;
+    _valid: boolean = true;
+    _required: boolean = false;
+    _value: string = '';
+
+    constructor() { }
+
+    @Input() get valid(): boolean { return this._valid; }
+
+    @Input() get disabled(): boolean { return this._disabled; }
+
+    @Input() get required(): boolean { return this._required; }
+
+    set required(isRequired) { this._required = anyToBoolean(isRequired); }
+
+    set disabled(isDisabled) { this._disabled = anyToBoolean(isDisabled); }
```

Comment:
You may also want to look into reactive/template based forms for setting disabled.